networkit/cpp/matching/test/MatcherGTest.cpp

```diff
@@ -279,4 +284,127 @@ TEST_F(MatcherGTest, testBSuitorMatcherDifferentB) {
     EXPECT_TRUE(M.isProper());
     EXPECT_FALSE(hasUnmatchedNeighbors(G, M));
 }
+
+TEST_F(MatcherGTest, testDynBSuitorInsertEdges) {
+    for (int i = 0; i < 10; i++) {
+        METISGraphReader graphReader;
+        auto G = graphReader.read("input/lesmis.graph");
+        std::vector<GraphEvent> events;
+        count m = 10;
+        // Select m edges of the graph, remove them but put them into edges for later insertion.
+        // This will make sure that the graph is valid.
+        for (count j = 0; j < m; j++) {
+            const auto [u, v] = GraphTools::randomEdge(G);
+            assert(G.hasEdge(u, v));
+            events.emplace_back(GraphEvent{GraphEvent::EDGE_ADDITION, u, v, G.weight(u, v)});
+            G.removeEdge(u, v);
+        }
+
+        const count b = 6;
+        DynamicBSuitorMatcher dbsm(G, b);
+        dbsm.run();
+
+        for (auto &e : events) {
+            G.addEdge(e.u, e.v, e.w);
+        }
+
+        dbsm.updateBatch(events);
+
+        dbsm.buildBMatching();
+        const auto dm = dbsm.getBMatching();
+        auto dres = dm.getMatches();
+        const auto dwm = dm.weight();
+
+        BSuitorMatcher bsm(G, b);
+        bsm.run();
+        bsm.buildBMatching();
```

Comment:
Since you need to compute static matches here and in other parts of the tests, consider implementing a helper function that tests the dynamic matching against a static one.